filesystem/android_super.ksy

```diff
@@ -0,0 +1,200 @@
+meta:
+  id: android_super
+  title: Android Dynamic Partitions metadata
+  application: Android
+  file-extension: img
+  license: CC0-1.0
+  ks-version: 0.9
+  bit-endian: le
+  endian: le
+
+doc: |
+  The metadata stored by Android at the beginning of a "super" partition, which
+  is what it calls a disk partition that holds one or more Dynamic Partitions.
+  Dynamic Partitions do more or less the same thing that LVM does on Linux,
+  allowing Android to map ranges of non-contiguous extents to a single logical
+  device. This metadata holds that mapping.
+
+doc-ref:
+  - https://source.android.com/devices/tech/ota/dynamic_partitions/implement
+  - https://android.googlesource.com/platform/system/core/+/master/fs_mgr/liblp/include/liblp/metadata_format.h
+
+seq:
+  - id: padding
+    size: 0x1000
+    type: dummy
```

Comment:
Is it needed?

Reply:
As I understand it, using a dummy type here creates a substream that's never consumed, so we skip over the bytes and never read them. If I omitted a type and just made it a byte array, we'd be reading and loading into memory the entire 4KB. It also adds a highlight in the Web IDE for all those 4KB, which I don't like since they're totally irrelevant and ignored.

Reply:
No. Substreams currently are read as bytes, then for each substream a KaitaiStream object created from bytes. So you should probably move everything except the padding into an instance.

Reply:
Updated to use instance.